File: app/src/main/java/eu/qm/fiszki/drawer/drawerItem/Exit.java
package eu.qm.fiszki.drawer.drawerItem;

import android.app.Activity;

import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import com.mikepenz.materialdrawer.model.interfaces.IconableKt;
import com.mikepenz.materialdrawer.model.interfaces.NameableKt;

import eu.qm.fiszki.R;

/**
 * Created by tm on 08.07.16.
 */
public class Exit extends PrimaryDrawerItem {

    public Exit(final Activity activity) {
        NameableKt.withName(this, R.string.drawer_exit_name);
        IconableKt.withIcon(this, R.drawable.ic_exit_to_app_black_24px);
        this.withOnDrawerItemClickListener((view, drawerItem, position) -> {
            activity.finish();
            return false;
        });
    }
}

File: app/src/main/java/eu/qm/fiszki/drawer/drawerItem/Exit.kt
package eu.qm.fiszki.drawer.drawerItem

import android.app.Activity
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem
import com.mikepenz.materialdrawer.model.interfaces.withIcon
import com.mikepenz.materialdrawer.model.interfaces.withName
import eu.qm.fiszki.R

class Exit(activity: Activity) : PrimaryDrawerItem() {

    init {
        withName(R.string.drawer_exit_name)
        withIcon(R.drawable.ic_exit_to_app_black_24px)
        withOnDrawerItemClickListener { _, _, _ ->
            activity.finish()
            false
        }
    }
}
